### PR TITLE
4.1.0

### PR DESCRIPTION
**Pull request by zzak**
2a813b3 is the meat of this release

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,12 +1,10 @@
-# -*- ruby -*-
-
-require 'rubygems'
 require 'hoe'
 
+SUPPORTED_VERSIONS = %w[1.8.7-p374 1.9.3-p551 2.0.0-p647 2.1.8 2.2.3 2.3.0]
```
Comment (hsbt) [inline Rakefile]: 2.2.4?

**Comment by zzak** [comment]
Here's the current blocker.. and it's got something to do with 1.8.

```
~ => ruby -v
ruby 1.8.7 (2013-12-22 patchlevel 375) [x86_64-linux]

~ => rdoc-data --install

~ => ri Object
/home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:548:in `load': undefined class/module Encoding (ArgumentError)
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:548:in `load_cache'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:547:in `open'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:547:in `load_cache'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:391:in `initialize'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/paths.rb:54:in `each'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:385:in `initialize'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:362:in `new'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:362:in `run'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/bin/ri:12
        from /home/zzak/.gem/ruby/1.8.7/bin/ri:23:in `load'
        from /home/zzak/.gem/ruby/1.8.7/bin/ri:23
```

**but** if I clear the ri cache.. it works:

```
ruby -rubygems -e 'require "rdoc"; require "rdoc/ri/paths"; system "rm #{RDoc::RI::Paths.system_dir}/cache.ri"'
```

**Comment by drbrain** [comment]
Did you build the data with Ruby 1.8.7? Can you dump the cache.ri with ri --dump

**Comment by zzak** [comment]
@drbrain I think just using the system rdoc, but can change that to use the same version of Ruby that we're building.

Also, `ri --dump` doesn't work for me:

```
ri --dump
/home/zzak/.rubies/1.8.7-p375/lib/ruby/1.8/optparse.rb:451:in `parse': missing argument: --dump (OptionParser::MissingArgument)
```

**Comment by drbrain** [comment]
You need to give it the file to dump (path to cache.ri)

**Comment by zzak** [comment]
@drbrain Almost ready to wrap this up, just doing some testing to make sure all versions are working properly. Then I should be able to ship the gem!

**Comment by zzak** [comment]
Now.. for some reason, the 1.8 dir isn't being included when building the gem.

```
~/code/rdoc-data => ls data/
1.8  1.9  2.0  2.1  2.2  2.3
~/code/rdoc-data => ls pkg/rdoc-data-4.1.0/data/
1.8  1.9  2.0  2.1  2.2  2.3
```

from the gem:

```
~/.gem/ruby/1.8.7/gems => ls rdoc-data-4.1.0/data/
1.9  2.0  2.1  2.2  2.3
```

**Pull request merged**